presto-main/src/main/java/com/facebook/presto/type/IpPrefixOperators.java

```diff
@@ -0,0 +1,246 @@
+    {
+        return left.compareTo(right) >= 0;
+    }
+
+    @ScalarOperator(BETWEEN)
+    @SqlType(StandardTypes.BOOLEAN)
+    @SqlNullable
+    public static Boolean between(@SqlType(StandardTypes.IPPREFIX) Slice value, @SqlType(StandardTypes.IPPREFIX) Slice min, @SqlType(StandardTypes.IPPREFIX) Slice max)
+    {
+        return min.compareTo(value) <= 0 && value.compareTo(max) <= 0;
+    }
+
+    @ScalarOperator(HASH_CODE)
+    @SqlType(StandardTypes.BIGINT)
+    public static long hashCode(@SqlType(StandardTypes.IPPREFIX) Slice value)
+    {
+        return XxHash64.hash(value);
+    }
+
+    @ScalarOperator(XX_HASH_64)
+    @SqlType(StandardTypes.BIGINT)
+    public static long xxHash64(@SqlType(StandardTypes.IPPREFIX) Slice value)
+    {
+        return XxHash64.hash(value);
+    }
+
+    @LiteralParameters("x")
+    @ScalarOperator(CAST)
+    @SqlType(StandardTypes.IPPREFIX)
+    public static Slice castFromVarcharToIpPrefix(@SqlType("varchar(x)") Slice slice)
+    {
+        byte[] address;
+        int subnetSize;
+        String string = slice.toStringUtf8();
+        if (!string.contains("/")) {
+            throw new PrestoException(INVALID_CAST_ARGUMENT, "Cannot cast value to IPPREFIX: " + slice.toStringUtf8());
+        }
+        String[] parts = string.split("/");
+        try {
+            address = InetAddresses.forString(parts[0]).getAddress();
+        }
+        catch (IllegalArgumentException e) {
+            throw new PrestoException(INVALID_CAST_ARGUMENT, "Cannot cast value to IPPREFIX: " + slice.toStringUtf8());
+        }
+        subnetSize = Integer.parseInt(parts[1]);
+
+        byte[] bytes = new byte[IPPREFIX.getFixedSize()];
+        if (address.length == 4) {
+            if (subnetSize < 0 || 32 < subnetSize) {
+                throw new PrestoException(INVALID_CAST_ARGUMENT, "Cannot cast value to IPPREFIX: " + slice.toStringUtf8());
+            }
+
+            for (int i = 0; i < 4; i++) {
+                address[3 - i] &= -0x1 << min(max((32 - subnetSize) - 8 * i, 0), 8);
+            }
+            bytes[10] = (byte) 0xff;
+            bytes[11] = (byte) 0xff;
+            arraycopy(address, 0, bytes, 12, 4);
+            bytes[IPPREFIX.getFixedSize() - 1] = (byte) subnetSize;
+        }
+        else if (address.length == 16) {
+            if (subnetSize < 0 || 128 < subnetSize) {
+                throw new PrestoException(INVALID_CAST_ARGUMENT, "Cannot cast value to IPPREFIX: " + slice.toStringUtf8());
+            }
+
+            for (int i = 0; i < 16; i++) {
+                address[15 - i] &= (byte) -0x1 << min(max((128 - subnetSize) - 8 * i, 0), 8);
+            }
+            arraycopy(address, 0, bytes, 0, 16);
+            bytes[IPPREFIX.getFixedSize() - 1] = (byte) subnetSize;
+        }
+        else {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, "Invalid InetAddrss length: " + address.length);
```

Comment:
I think ``GENERIC_INTERNAL_ERROR`` is reasonable here. If the input is invalid, ``InetAddresses.forString`` would throw, and the exception handling above would raise that as ``INVALID_CAST_ARGUMENT``. The throw here happens when ``InetAddress.getAddress`` returns a byte array with an unexpected length. Using ``GENERIC_INTERNAL_ERROR`` for that seems fine to me because it's the library that would've done something unexpected. 

Also, using ``GENERIC_INTERNAL_ERROR`` parallels the approach for similar logic in ``IpAddressOperators``.